### ISSUE TITLE
cftime has a problem with masked arrays

### ISSUE THREAD

**Issue by joernu76** [Unidata/cftime#211]
I am using cftime version "1.3.0" from anaconda.
I am using debian linux based on debian:latest updated to most recent packages.

We build a docker container, based on debian, install there anaconda and update everything to the most recent release to check for upcoming breakage.

The error occurs since about 14 days.
The following script:
```
import cftime
import numpy

a = [cftime.DatetimeGregorian(2014, 8, 1, 12, 0, 0, 0)]
b = numpy.asarray(a)
c = numpy.ma.asarray(a)
print(cftime.date2num(a, units="seconds since 2000-1-1"))                                      
print(cftime.date2num(b, units="seconds since 2000-1-1"))
print(cftime.date2num(c, units="seconds since 2000-1-1"))
```

has this output:
```
[4.602096e+08]
[4.602096e+08]
Traceback (most recent call last):
  File "a.py", line 9, in <module>
    print(cftime.date2num(c, units="seconds since 2000-1-1"))
  File "cftime/_cftime.pyx", line 222, in cftime._cftime.date2num
  File "/opt/conda/lib/python3.8/site-packages/numpy/ma/core.py", line 2661, in __getitem__
    result = self.dataiter.__getitem__(indx).view(type(self.ma))
AttributeError: 'cftime._cftime.DatetimeGregorian' object has no attribute 'view'
```

We expect all three print statements to give the number.

**Comment by jswhit** [comment]
Fix is in PR #212.  Problem was that `.flat[0]` doesn't work on masked arrays containing objects that don't have a `view` method.  Solution is to use `.item(0)` instead.

**Comment by joernu76** [comment]
Perfect! Thanks for the very quick fix. I look forward to see this appear also in anaconda soon!
